suite-common/trading/src/thunks/common/__tests__/watchTradeThunk.test.ts

```diff
@@ -0,0 +1,247 @@
+import { combineReducers } from '@reduxjs/toolkit';
+
+import { configureMockStore, extraDependenciesMock } from '@suite-common/test-utils';
+import { Account } from '@suite-common/wallet-types';
+
+import { accountBtc } from '../../../__fixtures__/utils';
+import { invityAPI } from '../../../invityAPI';
+import {
+    TradingState,
+    initialState,
+    prepareTradingReducer,
+} from '../../../reducers/tradingReducer';
+import {
+    TradingTransactionBuy,
+    TradingTransactionExchange,
+    TradingTransactionSell,
+} from '../../../types';
+import { watchTradeThunk } from '../watchTradeThunk';
+
+describe('watchTradeThunk', () => {
+    jest.mock('../../../invityAPI');
+
+    const tradingReducer = prepareTradingReducer(extraDependenciesMock);
+    const account = accountBtc as Account;
+    const accountData = {
+        symbol: account.symbol,
+        descriptor: account.descriptor,
+        accountIndex: account.index,
+        accountType: account.accountType,
+    };
+    const refreshCount = 1;
+
+    const getStore = (updatedState: Partial<TradingState>) =>
+        configureMockStore({
+            extra: {},
+            reducer: combineReducers({
+                wallet: combineReducers({
+                    tradingNew: tradingReducer,
+                }),
+            }),
+            preloadedState: {
+                wallet: {
+                    tradingNew: {
+                        ...initialState,
+                        ...updatedState,
+                    },
+                },
+            },
+        });
+
+    const date = new Date('2025-04-09');
+    const dateISO = date.toISOString();
+
+    beforeAll(() => {
+        jest.useFakeTimers();
+        jest.setSystemTime(date);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should remain in the same status when there is not any response', async () => {
+        const trade = {
+            date: dateISO,
+            key: 'tradeKey',
+            tradeType: 'buy',
+            data: {
+                status: 'LOGIN_REQUEST',
+                paymentId: 'tradeKey',
+            },
+            account: accountData,
+        } as TradingTransactionBuy;
+
+        const store = getStore({
+            trades: [trade],
+        });
+
+        invityAPI.watchTrade = () => Promise.resolve(undefined as any);
+
+        await store.dispatch(
+            watchTradeThunk({
+                account,
+                trade,
+                refreshCount,
+            }),
+        );
+
+        const actions = store.getActions();
+        const saveTradeAction = actions.find(action => action.type === '@trading/saveTrade');
+
+        expect(saveTradeAction).toBeUndefined();
+    });
+
+    it('status should stay same when status is the same', async () => {
```

Comment:
Please consider to rename it to something like this `should remain in the same status when there is same status in response` or better :)